src/infer_request.h

```diff
@@ -241,6 +247,7 @@ class InferenceRequest {
     std::vector<int64_t> shape_;
     std::vector<int64_t> shape_with_batch_dim_;
     bool is_shape_tensor_;
+    bool is_non_linear_format_io_;
```

Comment:
Can these two be combined as enum? i.e.
```
enum class {
  TENSOR;
  SHAPE_TENSOR;
  NON_LINEAR;
};

Reply:
Added a new enum class TensorType, encapsulated in Input class.
`enum class TensorType { TENSOR, SHAPE_TENSOR, NON_LINEAR };`